ospf6d/ospf6_gr.c

```diff
@@ -262,6 +262,22 @@ void ospf6_gr_restart_enter(struct ospf6 *ospf6,
 #define RTR_LSA_ADJ_FOUND 1
 #define RTR_LSA_ADJ_NOT_FOUND 2
 
+#define lsa_from_container(type, container) \
+	((struct type *)((char *)container->header + sizeof(struct ospf6_lsa_header)))
+
+#define lsdesc_start(lsdesc_type, lsa) \
```

Comment:
How about prefixing these macro names with "lsa_link_"? For example, "lsa_link_lsdesc_start(). While the naming in ospf6d is not ideal, no sense in making it even more cryptic.